Gordon360/Services/LostAndFoundService.cs

```diff
@@ -300,25 +300,74 @@ public IEnumerable<MissingItemReportViewModel> GetMissingItems(string requestedU
+        /// <param name="lastId">The ID of the last fetched report to start from for pagination</param>
+        /// <param name="pageSize">The size of the page to fetch for pagination</param>
         /// <returns>An enumerable of Missing Item Reports, from the Missing Item Data view</returns>
         /// <exception cref="UnauthorizedAccessException">If a user without admin permissions attempts to use</exception>
-        public IEnumerable<MissingItemReportViewModel> GetMissingItemsAll(string username)
+        public IEnumerable<MissingItemReportViewModel> GetMissingItemsAll(string username,
+                                                                          int? lastId,
+                                                                          int? pageSize,
+                                                                          string? status,
+                                                                          string? color,
+                                                                          string? category,
+                                                                          string? keywords)
         {
             if (!hasFullPermissions(username))
             {
                 throw new UnauthorizedAccessException();
             }
 
+            // Initialize database query to get all missing items ordered by date lost
+            IQueryable<MissingItemData> missingItems = context.MissingItemData.OrderByDescending(item => item.ID);
+
+            // Add filters to query based on provided filters
+            if (status is not null)
+            {
+                missingItems = missingItems.Where(x => x.status == status);
+            }
+            if (color is not null)
+            {
+                missingItems = missingItems.Where(x => x.colors.Contains(color));
+            }
+            if (category is not null)
+            {
+                missingItems = missingItems.Where(x => x.category == category);
+            }
+            if (keywords is not null)
+            {
+                missingItems = missingItems.Where(x => x.firstName.Contains(keywords)
+                                                    || x.lastName.Contains(keywords)
+                                                    || (x.firstName + " " + x.lastName).Contains(keywords)
+                                                    || x.description.Contains(keywords)
+                                                    || x.locationLost.Contains(keywords));
+            }
+
+            // Finally paginate filtered reports, based on the last ID the frontend received, and the size of the page to get
+            if (lastId is not null)
+            {
+                missingItems = missingItems.Where(item => item.ID < lastId);
+            }
+            if (pageSize is not null)
+            {
+                int pageLength = pageSize ?? default(int);
```

Comment:
`pageSize` can never be null here, so the right hand side of the null-coalescing `??` operator is unreachable. We can use pattern matching to assign the variable `pageLength` at the same time that we check that `pageSize` is not null:
```suggestion
            if (pageSize is int pageLength)
            {
```